### DISCUSSION TITLE
When I use "http://URL/dataset/page", the dataset page fails to display, showing "504-time out".

### DISCUSSION THREAD

**Discussion by hongjie123** [ckan/ckan#7027] [Q&A]
When I use "http://URL/dataset", the page works fine and the list of datasets is displayed successfully..
![image](https://user-images.githubusercontent.com/54384385/185568952-587bf9dc-0659-483b-9b07-72fc7af2404f.png)

But, When I use "http://URL/dataset/page", the dataset page fails to display, showing "504-time out".
![image](https://user-images.githubusercontent.com/54384385/185570984-ffc2e378-6f2a-4af1-aa1e-0a18d97a5afc.png)

**Comment by hongjie123** [comment]
I get the same error when I click on a link to a dataset. " 504 Time-out"

**Comment by Gauravp-NEC** [comment]
@hongjie123, I have use "http://url/dataset/page" and the dataset page shows "404 Not Found" which is a normal behavior in CKAN.

![image](https://user-images.githubusercontent.com/66116382/185845460-eba4d86e-084e-427a-b369-89525c060587.png)

**Comment by hongjie123** [comment]
@Gauravp-NEC , He means the URL of a specific data set. For example: http://47.93.14.246/dataset/ckgg.
![image](https://user-images.githubusercontent.com/54384385/185854509-163dceb1-e34d-4980-8373-19a4c76e210f.png)

**Comment by Gauravp-NEC** [comment]
@hongjie123 , You are getting "504 Time-out" due to you environment related issue and it is not related to CKAN.

**Comment by hongjie123** [comment]
@Gauravp-NEC But now I have hundreds of data sets that can't be opened, all showing "504 time-out." Isn't it the CKAN upgrade?

**Comment by hongjie123** [comment]
@Gauravp-NEC , The dataset page cannot be displayed.
![image](https://user-images.githubusercontent.com/54384385/185887593-0c332d30-a44c-4671-a975-308395335a66.png)

**Comment by hongjie123** [comment]
@Gauravp-NEC , I found the following error in the CKAN log file:
![image](https://user-images.githubusercontent.com/54384385/186088094-94171855-5dcc-4b5a-9e76-f6aeadb29797.png)
![image](https://user-images.githubusercontent.com/54384385/186088159-b3b16e9a-f1a5-499b-9fef-054bbff25bc7.png)
![image](https://user-images.githubusercontent.com/54384385/186088234-73482547-f892-4c4b-a90c-04acd15da761.png)
The errors in the three images above are connected and appear in one place.

**Comment by amercader** [comment]
@hongjie123 the 504 timeout error is an NGINX error saying that CKAN crashed and failed to respond. Looking at the errors that you pasted I see requests being made to `/api/v1/honor-point` and `/tool/deepke`, which are not CKAN URLs. These are probably requests being made by a custom plugin that you are using on the site. I'd suggest removing any custom plugins you are using on the site and see which one is causing the failure

**Comment by hongjie123** [comment]
@amercader Since the site was built several years ago by someone else, there is no documentation left. Would you please tell CKAN the architecture diagram of the framework or which services are called to each other? Thank you very much!

**Comment by hongjie123** [comment]
Does CKAN invoke other third party services? China's Aliyun company helped me check the website, and found that the program has been connecting to port 8080 of 113.31.104.113, but port 8080 of 113.31.104.113 is disconnected. The preliminary judgment is that the website calls the resources of third-party products, but the third-party IP cannot access. Result in access to final 504. May I ask if the IP: 113.31.104.113 belongs to the CKAN program itself?
